nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -405,7 +405,14 @@ void CollisionMonitor::process(const Velocity & cmd_vel_in, const std_msgs::msg:
   }
 
   // Points array collected from different data sources in a robot base frame
-  std::vector<Point> collision_points;
+  std::unordered_map<std::string, std::vector<Point>> sources_collision_points_map;
+
+  // Fill collision_points array from different data sources
+  for (std::shared_ptr<Source> source : sources_) {
```

Comment:
The source data population happens below, I don't think you should be adding a new duplicate but adjusting the old one, no? 

If you create the map element, you can pass in the reference to that vector to `getData()` so that it populates the map in-line rather than copying it twice (once from the source, another to the map).

```
  // Fill collision_points array from different data sources
  for (std::shared_ptr<Source> source : sources_) {
    if (source->getEnabled()) {
      if (!source->getData(curr_time, collision_points) &&  // <-- Right here populate an element of the map rather than the vector (that is no longer defined)
        source->getSourceTimeout().seconds() != 0.0)
      {
        action_polygon = nullptr;
        robot_action.polygon_name = "invalid source";
        robot_action.action_type = STOP;
        robot_action.req_vel.x = 0.0;
        robot_action.req_vel.y = 0.0;
        robot_action.req_vel.tw = 0.0;
        break;
      }
    }
  }
```

Reply:
It seems like this for loop should be deleted and the block of code below needs to be adjusted